pkg/liquidity-mining/contracts/gauges/StakelessGaugeCheckpointer.sol

```diff
@@ -263,6 +271,67 @@ contract StakelessGaugeCheckpointer is IStakelessGaugeCheckpointer, ReentrancyGu
         }
     }
 
+    function _getSingleBridgeCost(IStakelessGauge gauge) internal view returns (uint256) {
+        // Some versions of the stakeless gauges did not implement this interface, so we need to try / catch the call.
+        // In case the interface is not present, the cost is 0.
+        // Malicious contracts are ruled out at this stage: gauges shall be validated in external functions before
+        // reaching this point.
+        try gauge.getTotalBridgeCost() returns (uint256 cost) {
+            return cost;
+        } catch {
+            return 0;
+        }
+    }
+
+    function _getGaugeTypeTotalBridgeCost(string memory gaugeType, uint256 minRelativeWeight)
+        internal
+        view
+        returns (uint256 totalCost)
+    {
+        uint256 currentPeriod = _roundDownBlockTimestamp();
+        uint256 gaugesAmount = _gauges[gaugeType].length();
+        EnumerableSet.AddressSet storage gauges = _gauges[gaugeType];
+
+        for (uint256 i = 0; i < gaugesAmount; ++i) {
+            IStakelessGauge gauge = IStakelessGauge(gauges.unchecked_at(i));
+
+            if (_gaugeController.gauge_relative_weight(address(gauge), currentPeriod) < minRelativeWeight) {
+                continue;
+            }
+
+            uint256 gaugeBridgeCost = _getSingleBridgeCost(gauge);
+            // If one gauge is costless, the same should apply for all the gauges of the same type.
+            if (gaugeBridgeCost == 0) {
+                break;
+            }
+
+            // Cost per gauge might not be the same if gauges come from different factories, so we add each
+            // gauge's bridge cost individually.
+            totalCost += gaugeBridgeCost;
+        }
+    }
+
+    function _getGaugeTypesTotalBridgeCost(string[] memory gaugeTypes, uint256 minRelativeWeight)
+        internal
+        view
+        returns (uint256 totalCost)
+    {
+        for (uint256 i = 0; i < gaugeTypes.length; ++i) {
+            string memory gaugeType = gaugeTypes[i];
+            totalCost += _getGaugeTypeTotalBridgeCost(gaugeType, minRelativeWeight);
+        }
+    }
+
+    function _checkpointGaugesAboveRelativeWeight(string[] memory gaugeTypes, uint256 minRelativeWeight) internal {
+        uint256 currentPeriod = _roundDownBlockTimestamp();
+
+        for (uint256 i = 0; i < gaugeTypes.length; ++i) {
+            _checkpointGauges(gaugeTypes[i], minRelativeWeight, currentPeriod);
+        }
+
+        _returnLeftoverEthIfAny();
```

Comment:
Since this is called a couple places, and is really important, maybe make it an explicit modifier? Like:

```
modifier refundsEth {
   _;
   _returnLeftoverEthIfAny();
}

function checkpointMultipleGauges(...) ... refundsEth {... }
```